### ISSUE TITLE
Static pages in app directory are surprisingly slow (500ms+)

### ISSUE THREAD

**Issue by rijk** [vercel/next.js#43817]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:52 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8112
    Binaries:
      Node: 16.18.1
      npm: 8.19.2
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.7-canary.1
      eslint-config-next: 13.0.2
      react: 18.2.0
      react-dom: 18.2.0
``` 

### Link to the code that reproduces this issue

No repo but can make one

### To Reproduce

1. Go to https://amex.enterpriseinnovationtoolkit.com/
2. Click the "Learning"/"Toolkit"/"Community" links in the nav
3. You can experience a noticeable delay. Looking in the Network tab these requests typically take between 500ms to 1s to resolve, even though they are (very simple) static pages

### Describe the Bug

I've noticed static pages are a lot slower than I expected; between 500ms-1s. I haven't made a reproduction repo yet, I'm also not sure if this is the speed to be expected.

<img width="886" alt="image" src="https://user-images.githubusercontent.com/159500/206204373-ffcc668e-fe84-49e7-a400-80be21fb0b9b.png">

Proof it's static (the [organization] part comes from the subdomain via a rewrite):

<img width="444" alt="image" src="https://user-images.githubusercontent.com/159500/206204514-43adea46-e65b-40cf-817b-d3c77867bce4.png">

### Expected Behavior

I expected the performance to be better or at least equal to SSR + Hydration (pages directory), but right now it's really not.

### How are you deploying your application? (if relevant)

Vercel

**Comment by rijk** [comment]
Interesting thing is, on other pages it is as fast as I would expect, example: https://amex.enterpriseinnovationtoolkit.com/accelerating-innovation/hey-meet-the-new-editor and switch lessons in the nav on the left. These take <100ms to download, despite being more complex.

**Comment by rijk** [comment]
This seems to be fixed as I cannot reproduce anymore in 13.0.7.
